app/main.py

```diff
@@ -1,214 +1,115 @@
+class Knight:
+    def __init__(self, name: str, power: int, hp: int, armour: str,
+                 weapon: str, potion: str):
+        self.name = name
+        self.base_power = power
+        self.hp = hp
+        self.armour = armour or []
```

Comment:
The `armour` attribute is initialized with an empty list if not provided. However, this might lead to issues if `armour` is expected to be a dictionary or another type. Ensure that the default value aligns with how `armour` is used in the `calculate_protection` method.